apps/fxc-front/src/app/components/2d/path-element.ts

```diff
@@ -254,6 +250,63 @@ export class PathElement extends connect(store)(LitElement) {
     this.postScoreToHost(score);
   }
 
+  private computeScore(points: LatLon[]): Score {
+    const track: ScoringTrack = {
+      points: points.map((point, i) => {
+        return {
+          ...point,
+          alt: 0,
+          timeSec: i * 60,
+        };
+      }),
+      minTimeSec: new Date().getTime() / 1000,
+    };
+    const result = optimize({ track }, this.getLeague()).next().value;
+    const score = new Score({
+      circuit: this.getCircuitType(result),
+      distance: result.lengthKm * 1000,
+      multiplier: result.multiplier,
+      closingRadius: result.closingRadius ? result.closingRadius * 1000 : null,
+      indexes: result.solutionIndices,
+    });
+    // force the score as computed because of an unwanted side effect in constructor.
+    score.forcePoints(result.score);
+    return score;
+  }
+
+  private getLeague(): ScoringRules {
```

Comment:
Please move that out the component, i.e. in the `logic` folder

Reply:
code moved